typescript/packages/lookslike-high-level/src/recipes/prompts.ts

```diff
@@ -73,8 +75,12 @@ const addToPrompt = handler<{ prompt: string }, { title: string }>(
   }
 );
 
-export const prompt = recipe<{ title: string }>("prompt", ({ title }) => {
-  title.setDefault("abstract geometric art");
+const Title = z.object({
+  title: z.string().describe("Image generation prompt").default("abstract geometric art"),
+});
+type Title = z.infer<typeof Title>;
+
+export const prompt = recipe<Title>(Title, ({ title }) => {
```

Comment:
you should be able to do the last three lines under the hood.

IIRC it's something like
`function recipe<T extends z.ZodType>(schema: T, fn: (data: OpaqueRef<Required<z.infer<T>>>): RecipeFactory<T, ReturnType<typeof fn>>;`

Reply:
```
export function recipe<TSchema extends z.ZodTypeAny>(
  schema: TSchema,
  fn: (input: OpaqueRef<Required<z.infer<TSchema>>>) => Value<any>
): RecipeFactory<z.infer<TSchema>, ReturnType<typeof fn>>;
```

something like this works